src/tangd.c

```diff
@@ -184,19 +188,39 @@ rec(http_method_t method, const char *path, const char *body,
                       "\r\n%s", strlen(enc), enc);
 }
 
-static struct http_dispatch dispatch[] = {
-    { adv, 1 << HTTP_GET,  2, "^/+adv/+([0-9A-Za-z_-]+)$" },
-    { adv, 1 << HTTP_GET,  2, "^/+adv/*$" },
-    { rec, 1 << HTTP_POST, 2, "^/+rec/+([0-9A-Za-z_-]+)$" },
-    {}
-};
-
 #define DEFAULT_PORT 9090
 
+static struct http_dispatch s_dispatch[] = {
+        { adv, 1 << HTTP_GET,  2, NULL},
+        { adv, 1 << HTTP_GET,  2, NULL},
+        { rec, 1 << HTTP_POST, 2, NULL},
+        {}
+};
+
 static int
-process_request(const char *jwkdir, int in_fileno)
+process_request(const char *jwkdir, int in_fileno, const char *endpoint)
 {
-    struct http_state state = { .dispatch = dispatch, .misc = (char*)jwkdir };
+    char adv_endpoint[MAX_URL];
+    char adv2_endpoint[MAX_URL];
```

Comment:
A better name for these variables would be `adv_` and `adv_thp_`, instead of `adv2`, which does not mean much.